advisories/hackage/cabal-install/HSEC-2023-0015.md

```diff
@@ -0,0 +1,95 @@
+```toml
+[advisory]
+id = "HSEC-2023-0015"
+cwe = [672]
+keywords = ["hackage", "mitm", "supply-chain"]
+
+[[affected]]
+package = "cabal-install"
+cvss = "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:N/I:H/A:N"
+[[affected.versions]]
+introduced = "1.24.0.0"
+fixed = "3.10.2.0"
+
+[[references]]
+type = "REPORT"
+url = "https://github.com/haskell/cabal/issues/8918#issuecomment-1521096581"
+[[references]]
+type = "FIX"
+url = "https://github.com/haskell/cabal/commit/dcfdc9cffd74cade4e8cf3df37c5993413ffd30f"
+```
+
+# Summary
+
+A problem was recently discovered in `cabal-install`'s
+implementation of the Hackage Security protocol that would allow an
+attacker who was in possession of a revoked private key and who
+could perform a man-in-the-middle attack against Hackage to use the
+revoked key to deliver malicious packages. At this time, this is
+only a theoretical attack - no keys have been revoked. Release
+3.10.2.0 of `cabal-install` contains a fix for this bug, and we have
+contacted distributors of older versions (such as Linux
+distributions) with a patch that they can apply.
+
+# Background
+
+Hackage Security is an implementation of [The Update Framework],
+which is a design for a package repository that allows untrusted
+mirrors without undermining software supply-chain security. In
+particular, Hackage Security cryptographically guarantees the
+following properties:
+
+ * Mirrors of Hackage cannot change the contents of packages. This
+   prevents the insertion of malicious code.
+
+ * Mirrors cannot omit newer packages for more than a few days
+   without clients noticing. This ensures both that mirrors cannot
+   maliciously deny security updates, and that mistakes in their
+   configuration will be noticed.
+
+Hackage has a [key policy file][] that delegates authority to a
```

Comment:
I'm not sure it matters, but `[The Update Framework]` does not have an extra `[]`

Reply:
This is a fully lawful reference to https://github.com/haskell/security-advisories/pull/124/files#diff-67c9759c390b69f4c15040d1757b78e0cd35bc86900cd3e2c62347a1e069f338R80

Reply:
@blackheaven I didn't intend it (I prefer the explicit trailing `[]` - a [*collapsed reference link*](https://spec.commonmark.org/0.30/#collapsed-reference-link) in commonmark jargon), but both are legal constructions.